### PR TITLE
Parse `:host` pseudo-class selector

### LINKED ISSUE #1347: Add basic support for `:host` pseudo-class
When encountering a [`:host` pseudo-class](https://developer.mozilla.org/en-US/docs/Web/CSS/:host), Alfa just fails to parse it and discard the full rule. This has caused trouble for, e.g., a `:host, :root` selector where Alfa discards the full rule after failing to parse the selector and therefore doesn't apply the declaration to the `:root` element.

We should add basic support of `:host`, have it match nothing at first, but at least parse it and keep the declarations.

- [ ] Follow-up on DEV-14044

### PR DESCRIPTION
**Pull request by rcj-siteimprove**
Resolves #1347

Updated the pseudo class parser to parse the `:host` selector.

### CODE REVIEW

File: packages/alfa-selector/src/selector.ts [hunk 1]
```diff
@@ -1489,6 +1491,25 @@ export namespace Selector {
...
+
+    private constructor() {
+      super("host");
+    }
+
+    public matches(_element: Element): boolean {
+      // The shadow DOM is currently not supported so there should not be any matches.
+      // If support for the shadow DOM is added, then this should be implemented.
+      return false;
+    }
```
Comment (Jym77) [inline packages/alfa-selector/src/selector.ts:1510]:
```suggestion
```
We can just remove it as [the super-class already does so](https://github.com/Siteimprove/alfa/blob/04373c2c37feafdea0c8926d7ac1d69538eee368/packages/alfa-selector/src/selector.ts#L751-L753) I do not feel the comment is adding that much given that it is a fairly common pattern we have for selectors we don't really support.

File: packages/alfa-selector/test/selector.spec.tsx [hunk 2]
```diff
@@ -692,6 +692,20 @@ test(".parse() parses a named pseudo-class selector", (t) => {
...
 
+test(".parse() parses :host pseudo-class selector", (t) => {
+  const parseResult = parse(":host");
+
+  if (parseResult.isErr()) {
+    t.fail(parseResult.getErr());
+    return;
+  }
+
+  t.deepEqual(parseResult.get().toJSON(), {
```
Comment (Jym77) [inline packages/alfa-selector/test/selector.spec.tsx:703]:
```suggestion
  t.deepEqual(parse(":host").get().toJSON(), {
```
We don't really need the first test. If the test throw an exception, that's almost as good as failing. Other tests in this file seem to use this pattern.

**Review by Jym77 — CHANGES_REQUESTED** [review]
Simplifying a bit.

**Comment by Jym77** [comment]
!pr help

**Comment by siteimprove-builduser** [comment]
Hi there 👋 

 You have the following options: 
 - `!pr help` writes this comment
 - `!pr dedupe` Deduplicate packages
 - `!pr depgraph` Update dependency graph
 - `!pr document` Update packages documentation

File: packages/alfa-selector/src/selector.ts [hunk 3]
```diff
@@ -1489,6 +1491,20 @@ export namespace Selector {
...
+   */
+  export class Host extends Pseudo.Class<"host"> {
+    public static of(): Host {
+      return new Host();
+    }
+
+    private constructor() {
+      super("host");
+    }
+
```
Comment (rcj-siteimprove) [inline packages/alfa-selector/src/selector.ts:1505]:
```suggestion
    }
```

**Review by Jym77 — APPROVED** [review]

**Comment by rcj-siteimprove** [comment]
!pr document

**Comment by Jym77** [comment]
!pr version

**Comment by Jym77** [comment]
!pr version

**Comment by Jym77** [comment]
!pr version

**Comment by Jym77** [comment]
!pr version

**Comment by Jym77** [comment]
!pr version

**Comment by Jym77** [comment]
!pr version

**Comment by Jym77** [comment]
!pr version

**Comment by Jym77** [comment]
!pr document

**Review by Jym77 — APPROVED** [review]

**Pull request merged**